subworkflows/local/manipulate_damage.nf

```diff
@@ -0,0 +1,134 @@
+        ch_rescaled_index = params.fasta_largeref ? SAMTOOLS_INDEX_DAMAGE_RESCALED.out.csi : SAMTOOLS_INDEX_DAMAGE_RESCALED.out.bai
+
+        ch_rescaled_bams  = MAPDAMAGE2.out.rescaled.join(ch_rescaled_index)
+            .mix(ch_skip_rescale)
+    }
+
+    if ( params.run_pmd_filtering ) {
+        // TODO Add module to produce Masked reference from given references and bed file (with meta specifying the reference it matches)?
+        // if ( params.pmdtools_reference_mask) {
+        //     MASK_REFERENCE_BY_BED()
+        // }
+
+        ch_pmdtools_input = ch_input_for_damage_manipulation.multiMap {
+            ignore_me, meta, bam, bai, ref_meta, fasta ->
+                bam: [ meta, bam, bai ]
+                fasta: fasta
+        }
+
+        PMDTOOLS_FILTER( ch_pmdtools_input.bam, params.damage_manipulation_pmdtools_threshold, ch_pmdtools_input.fasta )
+        ch_versions       = ch_versions.mix( PMDTOOLS_FILTER.out.versions.first() )
+
+        SAMTOOLS_INDEX_DAMAGE_FILTERED( PMDTOOLS_FILTER.out.bam )
+        ch_versions       = ch_versions.mix( SAMTOOLS_INDEX_DAMAGE_FILTERED.out.versions.first() )
+        ch_filtered_index = params.fasta_largeref ? SAMTOOLS_INDEX_DAMAGE_FILTERED.out.csi : SAMTOOLS_INDEX_DAMAGE_FILTERED.out.bai
+
+        ch_pmd_filtered_bams = PMDTOOLS_FILTER.out.bam.join( ch_filtered_index )
+
+        SAMTOOLS_FLAGSTAT_FILTERED( ch_pmd_filtered_bams )
+        ch_pmd_filtered_flagstat = SAMTOOLS_FLAGSTAT_FILTERED.out.flagstat
+    }
+
+    if ( params.run_trim_bam ) {
+        if ( params.run_pmd_filtering ) {
+            ch_to_trim = ch_pmd_filtered_bams.map{
+                meta, bam, bai ->
+                [ meta, bam ]
+            }
+        } else {
+            ch_to_trim = ch_bam_bai.map {
+                meta, bam, bai ->
+                [ meta, bam ]
+            }
+        }
+
+        ch_trimbam_input = ch_to_trim.map {
+            meta, bam ->
+                trim_left  = meta.strandedness == 'single' ? ( meta.damage_treatment == 'none' ? params.damage_manipulation_bamutils_clip_single_stranded_none_udg_left  : ( meta.damage_treatment == 'half' ? params.damage_manipulation_bamutils_clip_single_stranded_half_udg_left  : 0 )) : ( meta.damage_treatment == 'none' ? params.damage_manipulation_bamutils_clip_double_stranded_none_udg_left  : ( meta.damage_treatment == 'half' ? params.damage_manipulation_bamutils_clip_double_stranded_half_udg_left  : 0 ))
+                trim_right = meta.strandedness == 'single' ? ( meta.damage_treatment == 'none' ? params.damage_manipulation_bamutils_clip_single_stranded_none_udg_right : ( meta.damage_treatment == 'half' ? params.damage_manipulation_bamutils_clip_single_stranded_half_udg_right : 0 )) : ( meta.damage_treatment == 'none' ? params.damage_manipulation_bamutils_clip_double_stranded_none_udg_right : ( meta.damage_treatment == 'half' ? params.damage_manipulation_bamutils_clip_double_stranded_half_udg_right : 0 ))
```

Comment:
Can remove the round brackets, fine to just chain `?` `:`

Reply:
with this many chains it becomes kinda illegible without them I felt. the parentheses helped me in debugging.
Maybe I can remove most, but keep the outermost set (strandedness check)

Reply:
Removed some